src/riak_kv_qry_compiler.erl

```diff
@@ -441,33 +448,54 @@ col_index_and_type_of(Fields, ColumnName) ->
             {Position, Type}
     end.
 
-expand_where(Where, #key_v1{ast = PAST}, MaxSubQueries) ->
+%%
+-spec expand_where(filter(), #key_v1{}, integer()) ->
+        [where_props()] | {error, any()}.
+expand_where(Where, PartitionKey, MaxSubQueries) ->
+    case find_quantum_field_index_in_key(PartitionKey) of
+        {QField, QSize, QUnit, QIndex} ->
+            hash_timestamp_to_quanta(QField, QSize, QUnit, QIndex, MaxSubQueries, Where);
+        notfound ->
+            [Where]
+    end.
+
+%% Return the parameters for the quantum function and it's index in the
+%% partition key fields.
+-spec find_quantum_field_index_in_key(#key_v1{}) ->
+    {QName::binary(), QSize::integer(), QUnit::atom(), QIndex::integer()} | notfound.
+find_quantum_field_index_in_key(#key_v1{ ast = PKAST }) ->
+    find_quantum_field_index_in_key2(PKAST, 1).
+
+%%
+find_quantum_field_index_in_key2([], _) ->
+    notfound;
+find_quantum_field_index_in_key2([#hash_fn_v1{ mod = riak_ql_quanta,
+                                               fn = quantum,
+                                               args = [#param_v1{name = [X]}, Y, Z] }|_], Index) ->
```

Comment:
I know this is a copy/paste - but it would be good to fix it with descriptive names for the tuple elements - QField, QSize, QUnit.